src/components/dialog/dialog.js

```diff
@@ -114,6 +115,16 @@ const VUIDialog = ({
         dialog.setAttribute('tabindex', 1);
         dialog.setAttribute('aria-hidden', false);
 
+        // Alerts don't have close buttons
+        if (hasCloseBtn && !isAlert) {
```

Comment:
Shouldn't this include `!isModal` too as I believe modal dialogs shouldn't have close buttons i.e. they can only be closed via a secondary action like *Cancel* and the `ESC` key? See:

> Escape - There should be a method to close the dialog without taking any action. This could be implemented via a cancel button which is keyboard accessible. It is recommended that a dialog also be cancelled by pressing the Escape key with focus on any item.—https://www.w3.org/TR/wai-aria-practices/#dialog_modal

Usually a Modal and Alert dialogs will have 2 actions, a primary and secondary, e.g.

**[Submit]** [Cancel]

^ which should be implemented by the consumer leaving this sort of close button more for your traditional "X" for a non-modal dialog, unless I'm missing something?